frame/support/src/dispatch.rs

```diff
@@ -25,18 +25,21 @@ pub use frame_metadata::{
 };
 pub use crate::weights::{
 	SimpleDispatchInfo, GetDispatchInfo, DispatchInfo, WeighData, ClassifyDispatch,
-	TransactionPriority, Weight, PaysFee,
+	TransactionPriority, Weight, PaysFee, PostDispatchInfo,
 };
-pub use sp_runtime::{traits::Dispatchable, DispatchError, DispatchResult};
+pub use sp_runtime::{traits::Dispatchable, DispatchError};
 pub use crate::traits::{CallMetadata, GetCallMetadata, GetCallName};
 
+pub type DispatchResult = sp_runtime::DispatchResult<PostDispatchInfo>;
+pub type DispatchErrorWithInfo = sp_runtime::DispatchErrorWithInfo<PostDispatchInfo>;
```

Comment:
1. I think it is fine to call this just `DispatchError`. Any error type can embed some underlying data. Also makes it a bit more symmetric with `DispatchResult`. Both are generic over the info while the name of only one reflects this. 

2.  Maybe you can do `pub type DispatchResult = sp_runtime::DispatchResult<DispatchErrorWithInfo>;` because now we can mistakingly do: 

```
pub type DispatchResult = sp_runtime::DispatchResult<A>;
pub type DispatchErrorWithInfo = sp_runtime::DispatchErrorWithInfo<B>;
```

then `DispatchResult` is `sp_runtime::DispatchErrorWithInfo<A>` while we re-export it here as `sp_runtime::DispatchErrorWithInfo<B>`.

Reply:
1. I agree. It would be nicer to drop that `WithInfo`suffix. There are some places in the code where the raw `DispatchError` is used. I look into it if it is feasible to change those places.

2. I do not understand what you are saying. Are you proposing that we should change `sp_runtime::DispatchResult` which is currently declared as follows?:
```
pub type DispatchResult<T> = Result<T, DispatchErrorWithInfo<T>>
```

Reply:
my suggestion was wrong, but the tricky situation is still valid and can happen. All in all, the problem is that `type DispatchErrorWithInfo` must be the same as `Err` variant of `DispatchResult ` which is not great since two types need to be the same without anything forcing them to do so (now back to my example with `A` and `B`). 

But it is not a big issue all in all.

Reply:
Regarding 1: I looked at the code and there are some places where one uses `DispatchError` for non-dispatch functions. Mostly helper functions. Something like this:
```
fn calc_me_something() -> Result<CalcResult, DispatchError> {
    Ok(complicated_calc()?)
}
```
It makes no sense to embed the `PostDispatchInfo` here as it would only be contained in the error path but not in the `Ok` path. The reason is that this is no real dispatch. The `DispatchError` is just abused in absence of a more specialized error.

When we now rename our type alias to `DispatchError` that would generate more confusion that it helps IMO. There are many places where we WANT the raw `DispatchError`. Maybe a rename to `DispatchErrorWithPostInfo`would be helpful.

Reply:
Now that I am thinking about it. I think we shouldn't have this type alias there after all. All the functions using `DispatchError`right now should either use something like `(MyAdditionalInfo, DispatchResult)` or use a `.map(|e| e.error)` to map away the `PostDispatchInfo`.

There is no reason to have it in the error case and not in the `Ok` case.

Reply:
While trying to remove `DispatchErrorWithPostInfo` I ran into some legitimate use cases where someone really want to name that type to store the error (in pallet_contracts). I come to the conclusion that the way it currently is should suffice and further changes will happen when pallets try to return the actual weight and must pass it through their helper function.